File: CS/ImageSliderWithWatermark/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace ImageSliderWithWatermark.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            ViewData["isWatermarkApplied"] = false;
            return View();
        }

        [HttpPost]
        public ActionResult Index(bool? applyWatermarkCheckBox)
        {
            ViewData["isWatermarkApplied"] = applyWatermarkCheckBox;
            return View();
        }
    }
}

File: VB/ImageSliderWatermark/Controllers/HomeController.vb
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        ViewData("isWatermarkApplied") = True
        Return View()
    End Function

    <HttpPost>
    Public Function Index(ByVal applyWatermarkCheckBox As Boolean?) As ActionResult
        ViewData("isWatermarkApplied") = applyWatermarkCheckBox
        Return View()
    End Function
End Class